src/probnum/diffeq/odefiltsmooth/initialize.py

```diff
@@ -0,0 +1,318 @@
+        InitialValueProblem object where the attribute `dy0_all` is filled with an
+        :math:`d(\nu + 1)` vector, where :math:`\nu` is the specified order,
+        and :math:`d` is the dimension of the IVP.
+
+
+    Examples
+    --------
+    >>> import sys, pytest
+    >>> if sys.platform.startswith('win'):
+    ...     pytest.skip('this doctest does not work on Windows')
+
+    >>> from probnum.problems.zoo.diffeq import threebody_jax, vanderpol_jax, threebody, vanderpol
+
+    Compute the initial values of the restricted three-body problem as follows
+
+    >>> res2bod = threebody_jax()
+    >>> print(res2bod.y0)
+    [ 0.994       0.          0.         -2.00158511]
+    >>> print(res2bod.dy0_all)
+    None
+    >>> res2bod = extend_ivp_with_all_derivatives(res2bod, order=3)
+    >>> print(res2bod.y0)
+    [ 0.994       0.          0.         -2.00158511]
+    >>> print(res2bod.dy0_all)
+    [ 9.94000000e-01  0.00000000e+00 -3.15543023e+02  0.00000000e+00
+      0.00000000e+00 -2.00158511e+00  0.00000000e+00  9.99720945e+04
+      0.00000000e+00 -3.15543023e+02  0.00000000e+00  6.39028111e+07
+     -2.00158511e+00  0.00000000e+00  9.99720945e+04  0.00000000e+00]
+
+    Compute the initial values of the van-der-Pol oscillator as follows
+
+    >>> vdp = vanderpol_jax()
+    >>> print(vdp.y0)
+    [2. 0.]
+    >>> print(vdp.dy0_all)
+    None
+    >>> vdp = extend_ivp_with_all_derivatives(vdp, order=3)
+    >>> print(vdp.y0)
+    [2. 0.]
+    >>> print(vdp.dy0_all)
+    [    2.     0.    -2.    60.     0.    -2.    60. -1798.]
+
+
+    >>> vdp2 = vanderpol()
+    >>> print(vdp2.y0)
+    [2. 0.]
+    >>> print(vdp2.dy0_all)
+    None
+    >>> vdp2 = extend_ivp_with_all_derivatives(vdp2, order=3)
+    >>> print(vdp2.y0)
+    [2. 0.]
+    >>> print(np.round(vdp2.dy0_all, 1))
+    [    2.      0.     -2.     58.2     0.     -2.     60.  -1743.4]
+    >>> print(np.round(np.log10(vdp2.dy0_errors), 1))
+    [ -inf -14.  -14.   -1.5  -inf -14.  -14.   -1.5]
+    """
+    try:
+        all_initial_derivatives, errors = compute_all_derivatives_via_taylormode(
+            f=ivp.f, z0=ivp.y0, t0=ivp.t0, order=order
+        )
+    except KeyError:
+        all_initial_derivatives, errors = compute_all_derivatives_via_rk(
+            f=ivp.f, z0=ivp.y0, t0=ivp.t0, order=order, df=ivp.df
+        )
+
+    return InitialValueProblem(
+        f=ivp.f,
+        t0=ivp.t0,
+        tmax=ivp.tmax,
+        y0=ivp.y0,
+        df=ivp.df,
+        ddf=ivp.ddf,
+        solution=ivp.solution,
+        dy0_all=all_initial_derivatives,
+        dy0_errors=errors,
+    )
+
+
+SMALL_VALUE = 1e-28
+
+
+def compute_all_derivatives_via_rk(f, z0, t0, order, df=None, h0=1e-2, method="DOP853"):
+    r"""Compute derivatives of the initial value by fitting an integrated Brownian motion process to a few steps of an approximate ODE solution.
+
+
+    It goes as follows:
+
+    1. The ODE integration problem is set up on the interval ``[t0, t0 + (2*order+1)*h0]``
```

Comment:
the choice 2q+1 is rather arbitrary. though I vaguely remember from some old Nordsieck and gear paper that q+1 steps are necessary. I found the performance to be improved with a few more steps than that, so 2q+1 is just for good measure.